lib/r10k/environment/base.rb

```diff
@@ -106,8 +129,12 @@ def info
   # @return [Array<R10K::Module::Base>] All modules defined in the Puppetfile
   #   associated with this environment.
   def modules
```

Comment:
note: this method is overidden by `Environment::WithModules` which Git environments inherit, and so is unused unless you're using SVN backed envs.

Reply:
Could this method be called before an environment.sync?